src/ui/screens/ProposeTipScreen.tsx

```diff
@@ -11,14 +11,21 @@ import {Padder} from '@ui/components/Padder';
 import {Account} from 'src/api/hooks/useAccount';
 import {useRefetch} from 'src/api/hooks/useRefetch';
 import {TIPS_QUERY} from 'src/api/hooks/useTips';
+import {InputLabel} from '@ui/library/InputLabel';
+import {isAddressValid} from 'src/utils/address';
+import {NetworkContext} from 'context/NetworkContext';
 
 export function ProposeTipScreen({navigation}: {navigation: NavigationProp<DashboardStackParamList>}) {
-  const {colors} = useTheme();
   const [state, dispatch] = useReducer(reducer, initialState);
   const startTx = useApiTx();
   const {refetch: refetchTips} = useRefetch([TIPS_QUERY]);
+  const {currentNetwork} = useContext(NetworkContext);
 
-  const valid = state.account && state.beneficiary && state.reason && state.reason.length > 4;
+  const isBeneficiaryAddressValid = useMemo(() => {
+    return state.beneficiary ? isAddressValid(currentNetwork, state.beneficiary) : false;
+  }, [state.beneficiary, currentNetwork]);
+
+  const valid = state.account && isBeneficiaryAddressValid && state.reason.length > 4;
```

Comment:
The new `AddressInput` component can encapsulate this logic. Just a suggestion, not meant to be included in this PR.